app/database.py

```diff
@@ -25,6 +25,24 @@ def getChallenge(post_id):
     challenge = db.session.query(Challenge).filter_by(id=post_id).first()
     return challenge
 
+def getChallengesByTitle(search):
+    challenges = db.session.query(Challenge).filter(Challenge.title.icontains(search)).all()
+    return challenges
+
+def getChallengesByCreator(search):
+    users = db.session.query(User).filter(User.username.icontains(search)).all()
+    ids = [user.id for user in users]
+    challenges = db.session.query(Challenge).filter(Challenge.creator_id.in_(ids)).all()
+    return challenges
+
+def getChallengesByTitleOrCreator(search):
```

Comment:
Seeing that you already have `getChallengesByTitle()` and `getChallengesByCreator()`, maybe you could just simply invoke those functions and combine the results in a list? Just make sure to remove any duplicate challenges.

Reply:
Well I made those two first, so I can remove them if everyones happy with searching by both title and creator name.

Reply:
I'm happy to say searching by both is a good idea

Reply:
The redundant getChallengesByTitle() and getChallengesByCreator() functions are removed now.